### PR TITLE
P4PU-64 sequence diagram ARC 

### PR DESCRIPTION
**Pull request by Giuseppe-LaManna**
#### List of Changes
- Added authentication dr sequences
- Added transactions list dr sequences
- Added transaction details dr sequences

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: design/arc/sequence_diagram_transactionDetails.mmd [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    deactivate User
+    activate BE_ARC
+        FE_ARC ->> BE_ARC : GET Transaction Details/<transactionId> <br> Authentication: Baerer <Session Token>
+        BE_ARC ->> BE_ARC : Validate session token
+    activate Biz_Events_Service
+    note right of BE_ARC : Retrieving CF from token
+        BE_ARC ->> Biz_Events_Service : Fiscal Code, Transaction Id
+    activate NDP
+    activate BE_ARC
+        Biz_Events_Service ->> NDP :
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionDetails.mmd]: missing description

File: design/arc/sequence_diagram_transactionDetails.mmd [hunk 2]
```diff
@@ -0,0 +1,30 @@
+sequenceDiagram
+actor User
+participant FE_ARC
+participant BE_ARC
+participant Biz_Events_Service
+participant NDP
+
+note right of User: User authenticated
+    activate User
+    activate FE_ARC
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionDetails.mmd:10]: please refactor the activation block to start from a call and end on the return

File: design/arc/sequence_diagram_transactionDetails.mmd [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+note over User, NDP : Retrieve Transaction Details
+        User ->> FE_ARC : Access to transaction details
+    deactivate User
+    activate BE_ARC
+        FE_ARC ->> BE_ARC : GET Transaction Details/<transactionId> <br> Authentication: Baerer <Session Token>
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionDetails.mmd]:
```suggestion
        FE_ARC ->> BE_ARC : GET Transaction Details/<transactionId> <br> Authentication: Bearer <Session Token>
```

**Review by and-mora — CHANGES_REQUESTED** [review]

File: design/arc/sequence_diagram_transactionsList.mmd [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    deactivate User
+    activate BE_ARC
+        FE_ARC ->> BE_ARC : GET Transactions List <br> Authentication: Baerer <Session Token>
+        BE_ARC ->> BE_ARC : Validate session token
+    activate Biz_Events_Service
+    note right of BE_ARC : Retrieving CF from token
+        BE_ARC ->> Biz_Events_Service : Fiscal Code
+    activate NDP
+    activate BE_ARC
+        Biz_Events_Service ->> NDP :
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionsList.mmd]: missing description

File: design/arc/sequence_diagram_transactionsList.mmd [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+participant Biz_Events_Service
+participant NDP
+
+note right of User: User authenticated
+    activate User
+    activate FE_ARC
+note over User, NDP : Retrieve Transactions List
+        User ->> FE_ARC : Access to transaction list
+    deactivate User
+    activate BE_ARC
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionsList.mmd]:
```suggestion
note over User, NDP : Retrieve Transactions List
        User ->> FE_ARC : Access to transaction list
    deactivate User
    activate BE_ARC
    activate FE_ARC
```
same as above about activation blocks

**Review by and-mora — CHANGES_REQUESTED** [review]

File: design/arc/sequence_diagram_authentication.mmd [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		FE_ARC ->> OneIdentity OpenId Provider : GET /<product>/login
+        activate OneIdentity OpenId Provider
+            OneIdentity OpenId Provider ->> OneIdentity OpenId Provider : save SAML response & create auth token
+            OneIdentity OpenId Provider ->> FE_ARC : return auth code and redirect to /token endpoint on product
+        deactivate OneIdentity OpenId Provider
+        FE_ARC -->> User : #
+    deactivate FE_ARC
+
+note over User, OneIdentity OpenId Provider: Token Request
+    User ->> FE_ARC : Auth token
```
Comment (and-mora) [inline design/arc/sequence_diagram_authentication.mmd]:
```suggestion
    User ->> FE_ARC : Auth code
```

File: design/arc/sequence_diagram_transactionDetails.mmd [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+participant FE_ARC
+participant BE_ARC
+participant Biz_Events_Service
+participant NDP
+
+note right of User: User authenticated
+note over User, NDP : Retrieve Transaction Details
+    User ->> FE_ARC : Access to transaction details
+    activate FE_ARC
+        FE_ARC ->> BE_ARC : GET Transaction Details/<transactionId> <br> Authentication: Bearer <Session Token>
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionDetails.mmd]:
```suggestion
        FE_ARC ->> BE_ARC : GET /transactions/<transactionId> <br> Authentication: Bearer <Session Token>
```

File: design/arc/sequence_diagram_transactionDetails.mmd [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        activate BE_ARC
+            BE_ARC ->> BE_ARC : Validate session token
+            note right of BE_ARC : Retrieving CF from token
+            BE_ARC ->> Biz_Events_Service : Fiscal Code, Transaction Id
+            activate Biz_Events_Service
+                Biz_Events_Service ->> NDP : retrieve user transaction details from the "nodo-dei-pagamenti"
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionDetails.mmd]: not sure to model NDP in this diagram. I'm not even sure it is correct: BizEvents reads the transactions list from its cosmos collection and not directly from NDP

File: design/arc/sequence_diagram_transactionsList.mmd [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+participant FE_ARC
+participant BE_ARC
+participant Biz_Events_Service
+participant NDP
+
+note right of User: User authenticated
+note over User, NDP : Retrieve Transactions List
+    User ->> FE_ARC : Access to transaction list
+    activate FE_ARC
+        FE_ARC ->> BE_ARC : GET Transactions List <br> Authentication: Bearer <Session Token>
```
Comment (and-mora) [inline design/arc/sequence_diagram_transactionsList.mmd]:
```suggestion
        FE_ARC ->> BE_ARC : GET /transactions <br> Authentication: Bearer <Session Token>
```

File: design/arc/sequence_diagram_transactionDetails.mmd (hunk 8 above)
Comment (and-mora) [reply design/arc/sequence_diagram_transactionDetails.mmd]: look at [DR](https://pagopa.atlassian.net/wiki/spaces/PPR/pages/872218824/Design+Review+Lista+Transazioni#Vista-dinamica-delle-componenti) for more insights.

**Review by and-mora — CHANGES_REQUESTED** [review]

File: design/arc/sequence_diagram_authentication.mmd [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    activate FE_ARC
+        FE_ARC ->> BE_ARC : /token
```
Comment (and-mora) [inline design/arc/sequence_diagram_authentication.mmd]:
```suggestion
        FE_ARC ->> BE_ARC : POST /token
```

**Review by and-mora — CHANGES_REQUESTED** [review]

File: design/arc/sequence_diagram_transactionDetails.mmd (hunk 8 above)
Comment (Giuseppe-LaManna) [reply design/arc/sequence_diagram_transactionDetails.mmd]: I can't see the link you passed, so I remove the NDP participant in this and in the  transactions list diagram?

Comment (and-mora) [reply design/arc/sequence_diagram_transactionDetails.mmd]: yes please

**Review by and-mora — APPROVED** [review]

**Pull request merged**